lib/layer.js

```diff
@@ -48,6 +48,31 @@ function Layer(path, options, fn) {
 }
 
 /**
+ * Handle any request for the layer.
+ *
+ * @api private
+ */
+
+Layer.prototype._handle_any = function _handle_any(/* ...args, next */) {
+  var self = this
+  var next = arguments[arguments.length - 1]
+  var handle = this.handle
+  var result
+  var hasNextBeenCalled = function () { return next._currentLayer === self }
+
+  try {
+    result = handle.apply(undefined, arguments)
```

Comment:
Ok, changed `undefined` to `(function () { return this })())` which works in theory. Here is my test:
```javascript
function strict() {
  'use strict';
  context.fn();
}

function sloppy() {
  context.fn();
}

var context = {
  a: 1,
  fn: function () {
    console.log(this.a);
    console.log((function () { return !!this })());
  }
};

strict();
sloppy();
```
and output:
```
1
true
1
true
```

Although I'm getting a different result from you (as you can see), both are returning the global object implying the `'use strict';` in `strict()` does nothing.